ext/IPSolverExt.jl

```diff
@@ -0,0 +1,104 @@
+module IPSolverExt
+
+import JuMP
+using ProblemReductions
+using LinearAlgebra
+
+function Base.findmin(problem::AbstractProblem, solver::IPSolver)
+    return _find(problem, solver,true)
+end
+function Base.findmax(problem::AbstractProblem, solver::IPSolver)
+    return _find(problem, solver,false)
+end
+
+# tag: true for min, false for max
+function _find(problem::AbstractProblem, solver::IPSolver,tag::Bool)
+    @assert num_flavors(problem) == 2 "IPSolver only supports boolean variables"
+    cons = constraints(problem)
+    nsc = ProblemReductions.num_variables(problem)
+    maxN = maximum([length(c.variables) for c in cons])
+    combs = [ProblemReductions.combinations(2,i) for i in 1:maxN]
+
+    objs = objectives(problem)
+    @assert all(length(obj.variables) <= 1 for obj in objs) "IPSolver only supports objectives with at most 1 variables"
+
+    # IP by JuMP
+    model = JuMP.Model(solver.optimizer)
+    !solver.verbose && JuMP.set_silent(model)
+
+    JuMP.@variable(model, 0 <= x[i = 1:nsc] <= 1, Int)
+    
+    for con in cons
+        f_vec = findall(!,con.specification)
+        num_vars = length(con.variables)
+        for f in f_vec
+            JuMP.@constraint(model, sum(j-> iszero(combs[num_vars][f][j]) ? (1 - x[con.variables[j]]) : x[con.variables[j]], 1:num_vars) <= num_vars -1)
+        end
+    end
+    
+    obj_sum = 0
+    for obj in objs
```

Comment:
Objectives can have more than one variables. The current code will scilently break for Spinglass.